core/src/test/java/org/apache/iceberg/hadoop/TestHadoopTables.java

```diff
@@ -55,20 +54,19 @@ public class TestHadoopTables {
           required(1, "id", Types.IntegerType.get(), "unique ID"),
           required(2, "data", Types.StringType.get()));
 
-  @Rule public TemporaryFolder temp = new TemporaryFolder();
   private File tableDir = null;
 
-  @Before
-  public void setupTableLocation() throws Exception {
-    tableDir = temp.newFolder();
+  @BeforeEach
+  public void setupTableLocation(@TempDir Path temp) throws Exception {
```

Comment:
the entire `setupTableLocation()` can be removed I believe

Reply:
Thank you for your review. I have now fixed the code based on your review and switched the file to JUnit5. I will review all your reviews and post a commit.